toolbox/main.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"istio.io/istio/tests/e2e/util"
+)
+
+var (
+	repo = flag.String("repo", "", "Update dependencies of only this repository")
+)
+
+func remote(repo string) (string, error) {
+	acct := *owner // passed in as flag in githubClient
+	token, err := getToken()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", acct, token, acct, repo), nil
```

Comment:
We discussed this in person but I shall repeat myself for the record of other contributors. Although cloning a git repo to local requires no authentication, pushing a new branch to the remote does.